lib/openai/runs.rb

```diff
@@ -0,0 +1,32 @@
+module OpenAI
+  class Runs
+    def initialize(client:)
+      @client = client.beta(assistants: "v1")
+    end
+
+    def list(thread_id:)
+      @client.get(path: "/threads/#{thread_id}/runs")
+    end
+
+    def retrieve(thread_id:, id:)
+      @client.get(path: "/threads/#{thread_id}/runs/#{id}")
+    end
+
+    def create(thread_id:, parameters: {})
+      @client.json_post(path: "/threads/#{thread_id}/runs", parameters: parameters)
+    end
+
+    def modify(id:, thread_id:, parameters: {})
+      @client.json_post(path: "/threads/#{thread_id}/runs/#{id}", parameters: parameters)
+    end
+
+    def cancel(id:, thread_id:)
+      @client.post(path: "/threads/#{thread_id}/runs/#{id}/cancel")
+    end
+
+    def submit_tool_outputs(thread_id:, run_id:, parameters: {})
+      @client.json_post(path: "/threads/#{thread_id}/runs/#{run_id}/submit_tool_outputs",
+                        parameters: parameters)
+    end
```

Comment:
perhaps we can figure this out in follow-up since the API itself is in BETA and shouldn't introduce any breaking changes. But the API ref documents essentially "nested" a step model inside of the runs model: https://platform.openai.com/docs/api-reference/runs/getRunStep ....it looks like those APIs are not exposed here in this iteration of the runs.rb ...in order to build these Assistants end-to-end we will need to get the Step Objects incorporated. (https://platform.openai.com/docs/api-reference/runs/step-object) .... but I don't have a good solution, just an observation that OPENAI didn't isolate steps into it's own class of API's...they just appended more API functionality to their runs model. Generally the PR LGTM. maybe I can start a PR once this goes in? or If you've pretty much have this going already lmk & I can just wait for those next round of changes. great work... 

Either way...the step objects are missing in this iteration? or maybe I'm just not seeing it. IMO this PR looks great and can be merged. In follow-up we need to add the step objects

Reply:
Good point - I've added support for them as a separate class.

```ruby
client.run_steps.list(thread_id: thread_id, run_id: run_id)
client.run_steps.retrieve(thread_id: thread_id, run_id: run_id, id: step_id)
```